### DISCUSSION TITLE
Vercel setup with yarn workspaces

### DISCUSSION THREAD

**Discussion by michelalbers** [vercel/vercel#4403] [Help]
Hi 👋 

I am working on a next.js application with a apollo node server as backend and I want to deploy them both via vercel. I am using a monorepo with yarn workspaces and typescript to do so.

This is my project structure

- root
  - packages
    - frontend (@myproject/frontend)
      - ... next.js app
    - api (@myproject/api)
      - src
        - index.ts and other sources in typescript
      - dist 
        - compiled api, not pushed to git
  - shared
    - store (@myproject/store)
      - src
        - models and other sources in typescript
      - lib
        - compiled version of the store - pushed to git

This is my now.json file:

```json
{
  "version": 2,
  "builds": [
    { "src": "packages/frontend/package.json", "use": "@now/next" },
    { "src": "packages/api/src/index.ts", "use": "@now/node" }
  ],
  "routes": [
    {
      "src": "/api/(.*)",
      "dest": "/packages/api/dist/index.js"
    },
    {
      "src": "(.*)",
      "dest": "/packages/frontend$1",
      "continue": true
    }
  ]
}

```

**Problem 1**
The next.js application works just fine via vercel but the api does not. When deploying the vercel compiler complains about not finding @myproject/store. 

**Problem 2**
I wanted to use ```now dev``` so I don't need to manually spin up the dev processes individually. That works for the next.js app but the api is ignored completely and not started.

What is it that I'm missing?

**Comment by aengl** [comment]
Not sure if that's the canonical solution, but the way I solved this problem is by publishing the shared dependency to a private GitHub package. In your `package.json` for `@myproject/store` you'd have something like:

```
{
  "name": "@myproject/store",
  "repository": {
    "type": "git",
    "url": "ssh://git@github.com/url/to/monorepo.git",
    "directory": "shared/store"
  },
  "publishConfig": {
    "registry": "https://npm.pkg.github.com/"
  },
  ...
}
```

Then publish it using `npm publish`. In the packages that depend on the store, you can then add a `.npmrc` to indicate where the packages are originated:

```
registry=https://registry.npmjs.org
@myproject:registry=https://npm.pkg.github.com
//npm.pkg.github.com/:_authToken=mytoken
```

You have to substitute `mytoken` with a token created at https://github.com/settings/tokens that has `read:packages` permissions.

Vercel will then deploy it without further configuration.

**Comment by paulogdm** [comment]
For the first problem, we don't support monorepos in a good way yet. The team is working on it as we speak. Allow me to say, however, that it may be a better idea to simply host the backend elsewhere since you may face limitations in our platform.

About the second problem: When using routes or builds, it may yield unexpected results. I recommend that you fallback to https://vercel.com/blog/zero-config when possible.

**Reply by belgattitude** [reply]
> The team is working on it as we speak

Really expecting this :smile:
